blocks/autocompleters/test/index.js

```diff
@@ -40,40 +46,55 @@ describe( 'blockAutocompleter', () => {
 
 	it( 'should prioritize common blocks in options', () => {
 		return blockAutocompleter( {} ).getOptions().then( ( options ) => {
-			// Exclude React element from label for assertion, since we can't
-			// easily test equality.
-			options = options.map( ( option ) => {
-				expect( option.label[ 0 ].type ).toBe( BlockIcon );
+			expect( options ).toMatchObject( [
+				blockTypes[ 'core/foo' ],
+				blockTypes[ 'core/baz' ],
+				blockTypes[ 'core/bar' ],
+			] );
+		} );
+	} );
 
-				return {
-					...option,
-					label: option.label.slice( 1 ),
-				};
-			} );
+	it( 'should render a block option label composed of @wordpress/element Elements and/or strings', () => {
+		expect.hasAssertions();
 
-			expect( options ).toEqual( [
-				{
-					keywords: [ 'keyword', 'foo' ],
-					label: [
-						'foo',
-					],
-					value: 'core/foo',
-				},
-				{
-					keywords: [ 'baz' ],
-					label: [
-						'baz',
-					],
-					value: 'core/baz',
-				},
-				{
-					keywords: [ 'bar' ],
-					label: [
-						'bar',
-					],
-					value: 'core/bar',
-				},
-			] );
+		const completer = blockAutocompleter( { } );
+		const renderedLabels = getBlockTypes().map(
+			blockType => completer.renderOptionLabel( blockType )
+		);
+
+		renderedLabels.forEach( label => {
+			// Only verify that a populated label is returned.
+			// It is likely to be fragile to assert that the contents are renderable by @wordpress/element.
+			if ( Array.isArray( label ) ) {
```

Comment:
Or can we make it more explicit what is what? At the moment it might work, but it's hard to decipher what we are checking here.

Reply:
Updated.